backend/src/ml_space_lambda/project/lambda_functions.py

```diff
@@ -123,18 +129,96 @@ def get(event, context):
         raise ResourceNotFound(f"Specified project {project_name} does not exist.")
     user = UserModel.from_dict(json.loads(event["requestContext"]["authorizer"]["user"]))
     project_user = project_user_dao.get(project_name, user.username)
-    if Permission.ADMIN not in user.permissions and not project_user:
+    is_member = is_member_of_project(user.username, project_name)
+    if Permission.ADMIN not in user.permissions and not project_user and not is_member:
         raise ValueError(f"User is not a member of project {project_name}.")
 
+    permissions = set(serialize_permissions(project_user.permissions) if project_user else [])
+    if is_member:
+        permissions.add(Permission.COLLABORATOR)
+    if is_owner_of_project(user.username, project_name):
+        permissions.add(Permission.PROJECT_OWNER)
+
     return {
         "project": project.to_dict(),
-        "permissions": serialize_permissions(project_user.permissions) if project_user else [],
+        "permissions": sorted(list(permissions)),
         "resourceCounts": (
             _get_resource_counts(project_name) if event["queryStringParameters"]["includeResourceCounts"] == "true" else {}
         ),
     }
 
 
+@api_wrapper
+def project_groups(event, context):
+    project_name = event["pathParameters"]["projectName"]
+    project = project_dao.get(project_name)
+    if not project:
+        raise ResourceNotFound(f"Specified project {project_name} does not exist.")
+
+    user = UserModel.from_dict(json.loads(event["requestContext"]["authorizer"]["user"]))
+    if Permission.ADMIN not in user.permissions and not is_member_of_project(user.username, project_name):
+        raise ValueError(f"User is not a member of project {project_name}.")
+
+    return [project_group.to_dict() for project_group in project_group_dao.get_groups_for_project(project_name)]
+
+
+@api_wrapper
+def add_groups(event, context):
+    project_name = event["pathParameters"]["projectName"]
+    request = json.loads(event["body"])
+    group_names = request["groupNames"]
+    for group_name in group_names:
+        group = group_dao.get(group_name)
+
+        if group:
```

Comment:
I wonder if we should check that the ProjectGroupModel doesn't exist for this relationship here too

Reply:
Good callout. We can skip it if it exists instead of erroring!